packages/block-editor/src/components/block-switcher/index.js

```diff
@@ -95,11 +95,22 @@ export const BlockSwitcherDropdownMenu = ( { clientIds, blocks } ) => {
 	const isTemplate = blocks.length === 1 && isTemplatePart( blocks[ 0 ] );
 
 	// Simple block tranformation based on the `Block Transforms` API.
-	const onBlockTransform = ( name ) =>
-		replaceBlocks( clientIds, switchToBlockType( blocks, name ) );
+	function onBlockTransform( name ) {
+		const newBlocks = switchToBlockType( blocks, name );
+		replaceBlocks( clientIds, newBlocks );
+		multiSelect(
+			newBlocks[ 0 ].clientId,
+			newBlocks[ newBlocks.length - 1 ].clientId
+		);
+	}
 	// Pattern transformation through the `Patterns` API.
-	const onPatternTransform = ( transformedBlocks ) =>
+	function onPatternTransform( transformedBlocks ) {
 		replaceBlocks( clientIds, transformedBlocks );
+		multiSelect(
```

Comment:
We might not want to do this for pattern transforms because patterns can contain many other blocks that doesn't necessarily provide a transformation back. This could be okay if the reasoning is to have more clarity about what has been inserted from a transformation.

Reply:
It's not just about having a transformation. It's about consistency too and as you say, to see what exactly has been inserted.